conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1311 @@
+    io:println(a1 is anydata[0]); // @output false
+
+    anydata a2 = [1, 2, 3];
+    io:println(a2 is anydata[]); // @output true
+    io:println(a2 is anydata[3]); // @output false
+
+    anydata a3 = [1, "foo", false];
+    io:println(a3 is anydata[]); // @output true
+    io:println(a3 is anydata[3]); // @output false
+
+    anydata[] a4 = [1, 2, 3];
+    io:println(a4 is anydata[]); // @output true
+    io:println(a4 is anydata[3]); // @output false
+
+    anydata|anydata[] a5 = [1, 2, 3];
+    io:println(a5 is anydata[]); // @output true
+    io:println(a5 is anydata[3]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a list-type-descriptor with anydata members.
+Labels: list-constructor-expr, is-expr, anydata, union-type, tuple-type, string, singleton-type
+
+function init() {
+    [anydata, anydata, anydata] t1 = [13, 25, "foo"];
+    io:println(t1 is [anydata, anydata, anydata]); // @output true
+    io:println(t1 is [13, 25, "foo"]); // @output false
+
+    [anydata, anydata...] t2 = [11, 27, "foo", "bar"];
+    io:println(t2 is [anydata, anydata...]); // @output true
+    io:println(t2 is [anydata, anydata, anydata, anydata]); // @output false
+    io:println(t2 is [11, 27, "foo", "bar"]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a list-type-descriptor of json.
+Labels: list-constructor-expr, is-expr, union-type, array-type, json, string
+
+function init() {
+    json a1 = [];
+    io:println(a1 is json[]); // @output true
+    io:println(a1 is json[0]); // @output false
+
+    json a2 = [true, 2, "foo"];
+    io:println(a2 is json[]); // @output true
+    io:println(a2 is json[3]); // @output false
+
+    json[] a3 = [true, 2, "foo"];
+    io:println(a3 is json[]); // @output true
+    io:println(a3 is json[3]); // @output false
+
+    json|json[] a4 = [true, 2, "foo"];
+    io:println(a4 is json[]); // @output true
+    io:println(a4 is json[3]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a list-type-descriptor of json.
+Labels: list-constructor-expr, is-expr, union-type, tuple-type, json, singleton-type, string
+
+function init() {
+    [json, json, json] t1 = [true, 2, "foo"];
+    io:println(t1 is [json, json, json]); // @output true
+    io:println(t1 is [true, 2, "foo"]); // @output false
+
+    [json, json...] t2 = [true, 2, "foo"];
+    io:println(t2 is [json, json...]); // @output true
+    io:println(t2 is [json, json, json]); // @output false
+    io:println(t2 is [true, 2, "foo"]); // @output false
+}
+
+Test-Case: output
+Description: Test inherent type of the created list when the applicable contextually expected type
+             is a list-type-descriptor with different types.
```

Comment:
Didn't get what's "different" in the tests? We can't have different types with just arrays?

Reply:
What I wanted to mean was something like "array-type-descriptor with different array member types". Updated the descriptions to make it obvious. Thanks!